### PR TITLE
nfd-worker: split out gRPC connection handling

### PR DESCRIPTION
**Pull request by marquiz**
Refactor the worker code and split out gRPC client connection handling
into a separate base type. The intent is to promote re-usability of code
for other NFD clients, too.

### CODE REVIEW

**Comment by marquiz** [comment]
This PR is targeting #525. There are other ways (probably better and more extensive) to make the code re-usable. But, this was a quick one that I came up with in the limited time I had.
/hold

File: pkg/nfd-client/base.go
Comment (ArangoGutierrez) [inline pkg/nfd-client/base.go:58]: I am getting 
```bash
I0804 19:27:31.330196       1 nfd-worker.go:167] NodeName: '%!s(func() string=0x8979c0)'
```
on the logs

Comment (marquiz) [reply pkg/nfd-client/base.go:58]: Well spotted! Fixed now

**Comment by marquiz** [comment]
Rebased.

Any comments @swatisehgal? This is targeting your work so we could merge this if you see it good and useful

**Comment by swatisehgal** [comment]
Thanks for the PR @marquiz. This change looks good to me!I have it already cherry-picked in my PR: https://github.com/kubernetes-sigs/node-feature-discovery/pull/525.

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

**Comment by marquiz** [comment]
> Thanks for the PR @marquiz. This change looks good to me!I have it already cherry-picked in my PR: #525.

Thanks @swatisehgal 
/unhold

**Comment by marquiz** [comment]
We can improve on the code re-use later on, if needed

**Pull request merged**
